### PR TITLE
✨ Create deployment template for helm chart

### LINKED ISSUE #171: Helm chart support for overriding images and image pull secrets
**User Story**

As a cluster fleet operator I would like to be able to use images from private registries in order to deploy the CAPI operator in an air gapped environment.

**Detailed Description**

In order to be able to allow a fully air gapped installation, the CAPI Operator helm chart should expose ways to override the images and image pull secrets used in the controller manager deployment at a minimum.

**Anything else you would like to add:**

Not sure if it is already tracked in https://github.com/kubernetes-sigs/cluster-api-operator/issues/130

/kind feature

### PR DESCRIPTION
**Pull request by Fedosin**
**What this PR does / why we need it**:
This PR makes capi operator deployment manifest customizable. E.g. you can set custom image or modify resource limits, and so on.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #171

### CODE REVIEW

**Comment by Fedosin** [comment]
/hold

need to update release image setting https://github.com/kubernetes-sigs/cluster-api-operator/blob/main/Makefile#L414-L418

File: Makefile [hunk 1]
```diff
@@ -374,11 +374,22 @@ set-manifest-pull-policy:
 	$(info Updating kustomize pull policy file for manager resources)
 	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' $(TARGET_RESOURCE)
 
+.PHONY: set-manifest-pull-policy-chart
+set-manifest-pull-policy-chart:
+	$(info Updating image pull policy value for helm chart)
+	yq eval '.image.manager.pullPolicy = "$(PULL_POLICY)"' $(TARGET_RESOURCE) -i
```
Comment (alexander-demicev) [inline Makefile:380]: can you make `yq` a local dependency like all other binaries we are using?

**Review by alexander-demicev — CHANGES_REQUESTED** [review]
thanks, looks good just one minor comment

**Review by alexander-demicev — COMMENTED** [review]
/approve

File: hack/charts/cluster-api-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    clusterctl.cluster.x-k8s.io/core: capi-operator
+    {{- with .Values.deploymentLabels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  {{- with .Values.deploymentAnnotations }}
+  annotations:
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.replicaCount }}
```
Comment (Danil-Grigorev) [inline hack/charts/cluster-api-operator/templates/deployment.yaml:21]: This value does not have default, please add it to values.yaml

File: hack/charts/cluster-api-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      {{- end }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
```
Comment (Danil-Grigorev) [inline hack/charts/cluster-api-operator/templates/deployment.yaml:145]: These values seems to be missing from values.yaml too

File: hack/charts/cluster-api-operator/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      {{- with .Values.volumes }}
+      volumes:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
```
Comment (Danil-Grigorev) [inline hack/charts/cluster-api-operator/templates/deployment.yaml:142]: Here as well

**Review by Danil-Grigorev — COMMENTED** [review]
Some missing defaults which I think need to be added, but overall looks good

**Review by Danil-Grigorev — APPROVED** [review]
/lgtm

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**